p2p/src/simulated/network.rs

```diff
@@ -163,6 +165,8 @@ impl<E: RNetwork + Spawner + Rng + Clock + Metrics, P: PublicKey> Network<E, P>
                         &mut self.context.clone(),
                         public_key.clone(),
                         self.get_next_socket(),
+                        egress_bps,
```

Comment:
Just took an initial scan here (SUPER cool ❤️ ).

Few Questions:
* How do you feel about putting these egress/ingress limits on simulated links? I'm torn on whether that seems more ~realistic for a simulated network model (since "channels" are an abstract concept to the Commonware Library it seems unnatural to put them there)?
* We could do both link limits AND aggregate peer limits (via a new function in `p2p::simulated` like `add_capacity(peer, ingress, egress)`? This would allow us to IMHO replicate any scenario we'd be interested in (as links may have a limit and peers via their switch may have a global limit)?

Thanks again for taking a spike...this is sick.

Reply:
My intuition was to add the limits per peer since I think that better models reality, i.e. the network bandwidth is shared among all the links / connections the peer has. With the current implementation, if you wanted to simulate a link capacity of say 100 Mbps, the actual aggregate bandwidth consumed by the peer would scale with the number of links it has (which doesn't seem realistic). I guess we could have an aggregate peer limit (as implemented now), and link limits which could only restrict the aggregate limit, but I wonder if that's very useful in practice though. For the estimator it could become very complicated to set it up, right now I'm already setting up the peer limits per region to ease on that.

Reply:
> I guess we could have an aggregate peer limit (as implemented now)

Ahhhhhhh, the way you implemented it via register confused me (this implies that there is a "limit" per channel):
<img width="570" height="667" alt="image" src="https://github.com/user-attachments/assets/4a37b872-f8a9-4762-bd52-c1e64cdd08f6" />

To set an "aggregate" limit, I recommend adding a standalone function called `limit` or `capacity` that takes a public key. That makes it clear that this spans multiple channels.

Reply:
(This approach of using `limit/capacity` has the convenient byproduct of not needing to be called by tests that do not care about this sort of thing).

Reply:
Updated the PR so that `register` doesn't have any bandwidth limits (same signature as before). There is now a `Oracle::set_bandwidth` method for this.